test/integration/tasks/delete-stale-entries.ts

```diff
@@ -0,0 +1,38 @@
+import { plainClient } from '../contentful-client'
+import { printStepTitle, sleep } from '../utils'
+
+const TWO_HOURS_IN_MS = 60 * 60 * 2 * 1000
+
+export default async (client = plainClient) => {
+  printStepTitle('Removing stale entries from master-test')
+
+  const entryCollection = await client.entry.getMany({ environmentId: 'master-test' })
+  const { items: entries } = entryCollection
+
+  const isStaleEntry = (timeStamp: string) => {
+    const entryDate = new Date(timeStamp).getTime()
+    const difference = Date.now() - entryDate
+    return difference >= TWO_HOURS_IN_MS
+  }
+
```

Comment:
You could consider using Promise.all here. I think the current approach is fine, but you called it out in the PR description. 

```
await Promise.all(entries.map(entry => {
 const {
      sys: { createdAt, id },
    } = entry

    if (!isStaleEntry(createdAt)) {
      return Promise.resolve()
    }
    try {
      await client.entry.delete({ entryId: id })
      console.log(`Deleted entry ${id}`)
      deletedEntries.push(id)
      await sleep(200)
    } catch (error) {
      console.error(`Could not delete entry ${id}`)
    }
}))
```

Reply:
We want to delete things sequentially here. That's also why we have the `sleep(200)` here (I think). We just had 800 stale entries in `master-test` and otherwise very certainly run into rate limiting.

Reply:
I agree with @Jwhiles here, maybe we can cleanup before merging this PR (e.g., run this version once, but keep this parallel for the upcoming runs. This might endup taking a long time should any problem with the test runner poop up

Reply:
Then I'll go for `Promise.allSettled` even though I think the risk of getting rate limited is higher than having to wait for a long time.

Reply:
Some requests would go through and in a couple of rounds we would be clean either way. So it sounds good to me. Test runs are already slow, wouldn't add much more overhead